src/app/models/followership.rb

```diff
@@ -0,0 +1,13 @@
+class Followership < ApplicationRecord
+  belongs_to :follower, class_name: :User
+  belongs_to :followee, class_name: :User
+
+  validates :followee, uniqueness: { scope: :follower }
+  validate :follower_and_followee
+
+  private
+
+  def follower_and_followee
+    errors.add(:follower, "User Can't Follow Himself") if follower_id == followee_id
```

Comment:
it's better to have this as a symbol and have the error string in locales, we did that for syntax error, checkout the locale/en.yml file
```suggestion
    errors.add(:follower, :cant_follow_self) if follower_id == followee_id
```